src/robusta/core/sinks/robusta/rrm/rrm.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import threading
+import time
+from datetime import datetime
+from typing import Optional
+
+from robusta.core.model.env_vars import RRM_PERIOD_SEC, MANAGED_CONFIGURATION_ENABLED
+from robusta.core.sinks.robusta.rrm.account_resource_fetcher import AccountResourceFetcher
+from robusta.core.sinks.robusta.rrm.prometheus_alert_resource_manager import PrometheusAlertResourceManager
+from robusta.core.sinks.robusta.rrm.types import ResourceKind, AccountResourceStatusType
+
+
+# robusta resource management
+class RRM:
+    def __init__(self, dal: AccountResourceFetcher, cluster: str, account_id: str):
+        self.dal = dal
+        self.cluster = cluster
+        self.__sleep = RRM_PERIOD_SEC
+        self.__last_updated_at: Optional[datetime] = None
+        self.__alert_resource_manager = PrometheusAlertResourceManager(
+            cluster=self.cluster, dal=dal,
+            resource_kind=ResourceKind.PrometheusAlert
+        )
+
+        self.__thread = threading.Thread(target=self.__thread_start)
+        self.__thread.start()
+
+    def __periodic_loop(self):
+        account_resources = self.dal \
+            .get_account_resources(updated_at=self.__last_updated_at)
+
+        for resource_kind, resources in account_resources.items():
```

Comment:
I would try to make this a little more generic:
```
errors: List[str] = []
for resource_kind, resources in account_resources.items():
  handler = resource_handlers.get(resource_kind)
  if handler:
    error = handler.handle_resources(resources)
    if error:
      errors.append(error)
  else:
    logger.warning(f"Could not handle resources from kind {resource_kind}")
 
 # then report sync status based on the collected errors   
    
```

Reply:
Done